txnkv/transaction/txn.go

```diff
@@ -763,6 +763,13 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tikv.LockCtx, keysInput
 			checkedExistence = true
 		}
 	}
+	if assignedPrimaryKey && lockCtx.LockOnlyIfExists && len(keys) == 1 {
+		if !txn.unsetPrimaryKeyIfNeed(lockCtx) {
+			memBuf.UpdateFlags(keys[0], tikv.SetKeyLocked, tikv.DelNeedCheckExists, tikv.SetKeyLockedValueExists)
+			txn.lockedCnt++
+		}
+		return nil
```

Comment:
Why not only unset the primary, and then do nothing but continue to the result handling and membuf updating logic below?
It seems when we are going to introduce other handling logic to these part (like adding some more flags), we will need to maintain both (L767-771 and L774-795). I'm afraid the new part will be very easily forgotten.

Reply:
In actually, I thought about doing that before.  I'll modify it.